jukepoot/puppet_parser.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+"""A parser of /var/log/puppet.log.
+
+
+"""
+
+
+import argparse 
+from datetime import datetime
+import dateutil.parser
+import re
+import collections
+import os
+
+
+def main():
+	parser = argparse.ArgumentParser()
+	parser.add_argument(
+		help="parse the given filename", type=str, 
+		dest="filename", action="store")
+	parser.add_argument(
+		"-y","--year", help="parse file by setting start year", 
+		type=int, dest="year", action="store")
+	args = parser.parse_args()
+	if args.year:
+		print("year to use: {0:d}".format(args.year))
+		year = args.year
+
+	else: #build year for date parsing
+		year = datetime.utcfromtimestamp(
+			os.stat(args.filename).st_birthtime).year
```

Comment:
I would prefer to split this a bit up... that is have something like

```
file_stat = os.stat(args.filename)
file_birth = file_stat.st_birthtime
year = datetime.datetime.utcfromtimestamp(file_birth).year
```

makes it a bit easier to read/debug